mint-client/src/clients/user.rs

```diff
@@ -420,3 +452,190 @@ impl From<LnClientError> for ClientError {
         ClientError::LnClientError(e)
     }
 }
+
+// -> clientd
+pub struct WrappedInvoice(pub lightning_invoice::Invoice);
+
+struct InvoiceVisitor {}
```

Comment:
Might be easier to build a serde module instead of visitor? See https://github.com/fedimint/minimint/blob/f5b3915b8545e458f8d948c93a0763c297a8be3a/mint-client/src/clients/gateway.rs#L419-L444 and https://github.com/fedimint/minimint/blob/f5b3915b8545e458f8d948c93a0763c297a8be3a/mint-client/src/clients/gateway.rs#L28-L29

Reply:
Ah yes, that makes sense. But since lightning_invoice::Invoice doesn't have a deserialize() method like SecretKey I could do String::deserialize(deserializer).parse::<lightning_invoice::Invoice>(); in the deserialize function right ?

Reply:
Yes, exactly, that's what I'd do. Maybe the serde module can be public and go into `modules/minimint-ln` so that I can also remove this shameful hack:

https://github.com/fedimint/minimint/blob/7cac6529e7e3c44b38399059080c977d334d9a7c/modules/minimint-ln/src/contracts/outgoing.rs#L23